parquet/src/main/scala/monix/connect/parquet/Parquet.scala

```diff
@@ -33,20 +47,19 @@ object Parquet {
     * @return A [[Consumer]] that expects records of type [[T]] to be passed and materializes to [[Long]]
     *         that represents the number of elements written.
     */
-  def writer[T](writer: ParquetWriter[T]): Consumer[T, Long] = {
+  @UnsafeBecauseImpure
+  def toWriterUnsafe[T](writer: ParquetWriter[T]): Consumer[T, Long] = {
```

Comment:
@Avasil any thoughts?

Reply:
https://github.com/monix/monix-connect/issues/352

Reply:
I think the naming `fromWriterUnsafe` would make sense here because we create `Consumer` from `ParquetWriter`

I would add "safe" versions (fromReader / fromWriter since reader/writer are deprecated?) that take either `Task` or at least by-name parameter and close resources.

Reply:
Followed your suggestions, on the other hand I guess that for the `ParquetSubscriber` will need to be reimplemented in such a way a way that takes as an argument a  `Task[ParquetWriter[T]]` instead of a raw `ParquetWriter[T]`, am I right?

Reply:
I'm not sure what you mean, it seems like it is the same situation as the reader

Reply:
It’s a bit different because in this case the subscriber will contain the logic to close resouces onComplete or onError. So we can not enclose it with any resouce data type. 
Therefore my idea is to change the current subscriber to expect a task with the parquet writer.

Reply:
It’s a bit different because in this case the subscriber will contain the logic to close resouces onComplete or onError. So we can not enclose it with any resouce data type. 
Therefore my idea is to change the current subscriber to expect a task with the parquet writer.

Reply:
Ah, I see...
Do we also want to do it in case of "shared" (strict parameter) `ParquetWriter`?

Reply:
Yeah why not? We are already providing a method that expects the strict parameter aka `def fromWriterUnsafe[T](parquetWriter: ParquetWriter[T]` and we want to make it compatible with  `def fromWriter[T](parquetWriter: Task[ParquetWriter[T]]`. 
I think would just have to implement a normal Subscriber instead of a Syncronous one, although in that case both will close resources on complete.

Reply:
Not sure if that would make any sense, will leave it as is for now.

Reply:
Added some tests, do you think this is ready to go?